UI/win-update/updater/updater.cpp

```diff
@@ -1085,6 +1086,72 @@ static bool UpdateFile(update_t &file)
 	return true;
 }
 
+queue<update_t *> updateQueue;
+static int lastPosition;
+static bool updateThreadFailed = false;
+
+static bool UpdateWorker()
+{
+	unique_lock<mutex> ulock(updateMutex, defer_lock);
+
+	while (true) {
+		ulock.lock();
+
+		if (updateThreadFailed)
+			return false;
+		if (updateQueue.empty())
+			break;
+
+		auto update = updateQueue.front();
+		updateQueue.pop();
+		ulock.unlock();
+
+		if (!UpdateFile(*update)) {
+			updateThreadFailed = true;
+			return false;
+		} else {
+			ulock.lock();
+			size_t updatesInstalled =
+				completedUpdates - updateQueue.size();
+			int position = (int)(((float)updatesInstalled /
+					      (float)completedUpdates) *
+					     100.0f);
+			if (position > lastPosition) {
+				lastPosition = position;
+				SendDlgItemMessage(hwndMain, IDC_PROGRESS,
+						   PBM_SETPOS, position, 0);
+			}
+			ulock.unlock();
```

Comment:
The locks here are *probably* not necessary but I wanted to be on the safe side.

Reply:
Removed the locks and avoided accessing the queue (even though `size()` is probably fine). `SendMessage` is supposed to be thread safe and duplicate progress messages probably aren't an issue worth worrying about.